src/lib/scene-internal.ts

```diff
@@ -662,6 +662,13 @@ export class SceneInternal implements Renderer {
     this.queueDraw();
   }
 
+  /**
+   * Wrap lookups for devicePixelRatio to satisfy aggressive compilation.
+   */
+  private get devicePixelRatio(): number {
+    return typeof window !== 'undefined' ? window.devicePixelRatio || 1 : 1;
```

Comment:
why not "window != null" ?

Reply:
If the `window` object is not defined in scope, I believe `window != null` will throw a `ReferenceError`.

This method of checking for `window` and `window.devicePixelRatio` is also used by vega: https://github.com/vega/vega/blob/main/packages/vega-scenegraph/src/util/canvas/resize.js#L2